dataland-frontend/src/components/forms/CreateEuTaxonomyForFinancials.vue

```diff
@@ -0,0 +1,655 @@
+    JumpLinksSection,
+    FailedUpload,
+    FormKit,
+    SuccessUpload,
+    UploadFormHeader,
+    Card,
+
+    UploadReports,
+    BasicInformationFields,
+
+    PrimeButton,
+    Calendar,
+    MultiSelect,
+    KPIfieldSet,
+  },
+
+  data() {
+    return {
+      formInputsModel: {} as CompanyAssociatedDataEuTaxonomyDataForFinancials,
+      filesToUpload: [] as ExtendedFile[],
+      listOfUploadedReportsInfo: [] as ExtendedCompanyReport[],
+      fiscalYearEndAsDate: null as Date | null,
+      fiscalYearEnd: "",
+      reportingPeriod: new Date(),
+      assuranceData: {
+        None: humanizeString(AssuranceDataAssuranceEnum.None),
+        LimitedAssurance: humanizeString(AssuranceDataAssuranceEnum.LimitedAssurance),
+        ReasonableAssurance: humanizeString(AssuranceDataAssuranceEnum.ReasonableAssurance),
+      },
+      euTaxonomyPseudoModelAndMappings,
+      euTaxonomyKpiNameMappings,
+      euTaxonomyKpiInfoMappings,
+      checkCustomInputs,
+      formatBytesUserFriendly,
+      updatePropertyFilesUploaded,
+      route: useRoute(),
+      waitingForData: false,
+      editMode: false,
+
+      postEuTaxonomyDataForFinancialsProcessed: false,
+      messageCount: 0,
+      postEuTaxonomyDataForFinancialsResponse: null as AxiosResponse<DataMetaInformation> | null,
+      humanizeString: humanizeString,
+      onThisPageLinksStart: [
+        { label: "Upload company reports", value: "uploadReports" },
+        { label: "Basic information", value: "basicInformation" },
+        { label: "Assurance", value: "assurance" },
+        { label: "Add KPIs", value: "addKpis" },
+      ],
+      onThisPageLinks: [] as { label: string; value: string }[],
+      kpisModel: [
+        { label: "Credit Institution", value: "creditInstitutionKpis" },
+        { label: "Investment Firm", value: "investmentFirmKpis" },
+        { label: "Insurance & Re-insurance", value: "insuranceKpis" },
+        { label: "Asset Management", value: "assetManagementKpis" },
+      ],
+      selectedKPIs: [] as { label: string; value: string }[],
+      confirmedSelectedKPIs: [] as { label: string; value: string }[],
+    };
+  },
+  computed: {
+    namesOfAllCompanyReportsForTheDataset(): string[] {
+      const namesFromFilesToUpload = this.filesToUpload.map((el) => el.name.split(".")[0]);
+      const namesFromListOfUploadedReports = this.listOfUploadedReportsInfo.map((el) => el.name.split(".")[0]);
+      return [...new Set([...namesFromFilesToUpload, ...namesFromListOfUploadedReports])];
+    },
+    reportingPeriodYear(): number {
+      return this.reportingPeriod.getFullYear();
+    },
+    computedFinancialServicesTypes(): string[] {
+      return this.confirmedSelectedKPIs.map((el: { label: string; value: string }): string => {
+        return euTaxonomyPseudoModelAndMappings.companyTypeToEligibilityKpis[
+          el.value as keyof typeof euTaxonomyPseudoModelAndMappings.companyTypeToEligibilityKpis
+        ];
+      });
+    },
+  },
```

Comment:
I change v-model for :ModelValue now value is read-only

Reply:
Reminder for myself:  Validate that it works when it is a read-only.

Reply:
Update: This is no computed property anymore => therefore this conversation is depricated:

BUT (!) =>  We should definitely rename the variable now, because it contains the word "computed".

Reply:
Another note: I have the feeling that it might be obsolete. Investigate.

Reply:
Fixed through refactoring.